MCUboot/README.md

```diff
@@ -1,163 +1,52 @@
-   cd ../application && mbed compile -t GCC_ARM -m NRF52840_DK
-   ```
-
-1. Copy the HEX file into the bootloader folder:
-
-   ```shell
-   cp BUILD/NRF52840_DK/GCC_ARM/application.hex ../bootloader && cd ../bootloader
-   ```
-
-1. Sign the initial application using the rsa-2048 keys:
-
-   ```shell
-   mcuboot/scripts/imgtool.py sign -k signing-keys.pem --align 4 -v 0.1.0 --header-size 4096 --pad-header -S 0xC0000 --pad application.hex signed_application.hex
-   ```
-   
-#### Creating and flashing the "factory firmware"
-
-1. Merge the bootloader and signed application:
-
-   ```shell
-   hexmerge.py -o merged.hex --no-start-addr BUILD/NRF52840_DK/GCC_ARM/bootloader.hex signed_application.hex
-   ```
-
-1. Erase the chip:
-   
-   ```shell
-   pyocd erase --chip
-   ```
-   
-1. Flash the "factory firmware" onto the board by copying `merged.hex` over to its mount point:
-
-   ```shell
-   cp merged.hex <mount_point>
-   ```
-   
-   The target's mount point can be found by running `mbedls`.
-   
-#### Creating the update binary
-
-1. Change the application's version number in its [mbed_app.json](target/application/mbed_app.json) to 0.1.1 and rebuild it:
-
-   ```shell
-   cd ../application && mbed compile -t GCC_ARM -m NRF52840_DK
-   ```
-
-1. Copy the HEX file into the bootloader folder:
-
-   ```shell
-   cp BUILD/NRF52840_DK/GCC_ARM/application.hex ../bootloader && cd ../bootloader
-   ```
-
-1. Sign the update application using the rsa-2048 keys:
-
-   ```shell
-   mcuboot/scripts/imgtool.py sign -k signing-keys.pem --align 4 -v 0.1.1 --header-size 4096 --pad-header -S 0x55000 application.hex signed_update.hex
-   ```
-
-1. Generate a raw binary file from `signed_update.hex` so that it can be transported over BLE:
-
-   ```shell
-   arm-none-eabi-objcopy -I ihex -O binary signed_update.hex signed_update.bin
-   ```
-
-### Demonstration
-
-1. Open a serial terminal on the host PC to receive serial prints from the target:
+Please make sure to activate the virtual environment, located in the root folder, before proceeding further. Also, ensure that the target board is connected and has been flashed with the factory firmware from the build stage.
 
+1. Open a serial terminal on the host computer to receive serial prints from the _FOTA target_:
    ```shell 
    mbed term -b 115200
    ```
-
-2. In a separate window, run the client script:
-
+2. In a separate shell window, run the test script from the client directory:
    ```shell
-   python client/client.py
+   python client.py
    ```
 
-   It scans for a device named 'FOTA' and attempts to connect to it.
+   It scans for a device named _"FOTA"_ and attempts to connect to it.
    Once connected, it asks the user to enter the path to the binary.
-   Enter the signed update binary:
+   Please enter the path to the signed update binary.
 
    ```
-   Enter the path to the binary: target/bootloader/signed_update.bin
+   Enter the path to the binary: ../target/bootloader/signed_update.bin
```

Comment:
why the added .. ? is target outside of the MCUboot direcotry?

Reply:
Nope, the target is still located in the mcuboot directory - I haven't touched the directory structure for both examples. Since the script is invoked from the client directory, the end-user would need to provide a relative path to the target - hence the `..`.